python/ray/data/_internal/planner/exchange/interfaces.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from ray.data._internal.execution.interfaces import RefBundle
+from ray.data._internal.stats import StatsDict
+from ray.data.block import Block, BlockMetadata
+
+
+class ExchangeInterface:
```

Comment:
This is copied from [ShuffleOp](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/shuffle.py#L9).

Decide to rename `ShuffleOp` to `ExchangeInterface`, because (1). `Op` name is overloaded (we now have logical and physical operator), and (2). avoid confusion as subclass implementations are shuffle, sort and groupby (not only shuffle).